common-pg/src/main/scala/com/socrata/pg/analyzer2/FuncallSqlizer.scala

```diff
@@ -147,13 +163,45 @@ abstract class FuncallSqlizer[MT <: MetaTypes] extends SqlizerUniverse[MT] {
     }
   }
 
-  protected def sqlizeNormalOrdinaryFuncall(sqlFunctionName: String, prefixArgs: Seq[Doc] = Nil, suffixArgs: Seq[Doc] = Nil) = {
+  def sqlizeNormalOrdinaryFuncall(
+    sqlFunctionName: String,
+    prefixArgs: Seq[Doc] = Nil,
+    suffixArgs: Seq[Doc] = Nil,
+    castType: (Doc, Int) => Option[Doc] = (_, _) => None
+  ) = {
     val funcName = Doc(sqlFunctionName)
     ofs { (e, args, ctx) =>
       assert(args.length >= e.function.minArity)
       assert(e.function.allParameters.startsWith(args.map(_.typ)))
 
-      val sql = (prefixArgs ++ args.map(_.compressed.sql) ++ suffixArgs).funcall(funcName)
+      val castArgs = args.map(_.compressed.sql).zipWithIndex.map { case (arg, idx) =>
+        castType(arg, idx).map { cast =>
+          arg +#+ Doc("::") +#+ cast
+        }.getOrElse(arg)
+      }
+
+      val argsSql = (prefixArgs ++ castArgs ++ suffixArgs)
+
+      val sql = argsSql.funcall(funcName)
+
+      ExprSql(sql.group, e)
+    }
+  }
+
+  def binaryOpCallFunctionPrefix(
```

Comment:
This is a weird function, though it's only weird because SQL is weird. :)  - I think it belongs in the SoQLFunctionSqlizerRedshift class (because in the PG class I'd insist on any special-syntax function get wrapped up in a stored proc.  I'm not super fond of the name but I don't really have any better suggestion for it off the top of my head.